dcapp/app/src/main/java/com/example/michael/mqtttest/BroadcastService.java

```diff
@@ -0,0 +1,239 @@
+     *         the service is active.
+     *
+     * @return - This method returns 'STAR_STICKY' to indicate that the the
+     *         system should attempt to recreate the service if the process is
+     *         killed.
+     */
+    public int onStartCommand(Intent intent, int flags, int startId) {
+
+        broadcastServiceIsRunning = true;
+
+        // Intent becomes null when DCAPP closes. Need to make sure intent is
+        // not null to prevent crash from accessing unset extras
+        androidId = (intent != null) ?
+                intent.getStringExtra(DEVICE_ID_INTENT_EXTRA) :
+                NO_ANDROID_ID_FOUND;
+
+        hostIp = (intent != null) ?
+                intent.getStringExtra(HOST_IP_INTENT_EXTRA) :
+                NO_IP_ADDRESS_FOUND;
+
+        delayValue = (intent != null) ?
+                intent.getIntExtra(SPEED_SETTING_INTENT_EXTRA, 0) :
+                NO_USER_SPEED_INPUT;
+
+        client = new MqttAndroidClient(this, TCP_PREFIX + hostIp, androidId);
+
+        sensorManager = (SensorManager) getApplicationContext()
+                .getSystemService(SENSOR_SERVICE);
+        accelerometer = sensorManager.getDefaultSensor(Sensor.TYPE_ACCELEROMETER);
+        accelerationListener = new AccelerationListener();
+        sensorManager.registerListener(accelerationListener,
+                accelerometer,
+                SensorManager.SENSOR_DELAY_NORMAL);
+
+        return START_STICKY;
+    }
+
+    @Override
+    /**
+     * @desc - This method is called every time BroadcastService is
+     *         destroyed. When this occurs, the broadcastServiceIsRunning flag
+     *         should be disabled and the accelerationListener unregistered.
+     */
+    public void onDestroy() {
+        broadcastServiceIsRunning = false;
+
+        sensorManager.unregisterListener(accelerationListener);
+        super.onDestroy();
+    }
+
+    /**
+     * @desc - This class is responsible for ensuring a connection is
+     *         established with the MQTT broker and publishing data to the
+     *         MQTT broker as a JSON on each acceleration event. Delays
+     *         between broadcasts are calculated based on input from the
+     *         user which will kill the function before publishing data
+     *         if insufficient time has elapsed.
+     *
+     *         NOTE: See description for BroadcastService above. We should
+     *         change this in the future.
+     */
+    public class AccelerationListener implements SensorEventListener {
+
+        private SensorEvent event;
+
+        @Override
+        public void onSensorChanged(SensorEvent event) {
+            this.event = event;
+
+            if (event.sensor.getType() == Sensor.TYPE_ACCELEROMETER) {
+                // Kill function if not enough time has elapsed between
+                // acceleration events (Should re-work this)
+                if(tickAccumulator++ % (delayValue * 5 + 1) != 0) {
```

Comment:
it's probably better to move the `tickAccumulator++` statement out to its own line. Also, the tick seems to be a lousy way to measure time and determine how often to do things.